src/pro/escape_special_char.pro

```diff
@@ -1,84 +1,108 @@
+; code in the future ... All "file_*" functions should be able to
+; manage by them-selves the special chars.
+;
 ; ESCAPE_SPECIAL_CHAR function escape special characteres present
 ; in input string argument and returns the result.
 ; Special characteres can be given by Keyword : list_of_special_char
 ; or default list is used.
 ;
-; MODIFICATION HISTORY:
-;   - June 2010: created by Lea Noreskal
-;   - 28 June 2010: modified by Alain Coulais (N_PARAMS(), /help)
-;-
 ; LICENCE:
 ; Copyright (C) 2010, Lea Noreskal and Alain Coulais
 ; This program is free software; you can redistribute it and/or modify  
 ; it under the terms of the GNU General Public License as published by  
 ; the Free Software Foundation; either version 2 of the License, or     
 ; (at your option) any later version.
-;-
 ;
-; ---------------------------------
+; AC 2018 : we do removed "." in the list 
+; but not sure if "?" or "*" should be in the list ...
+;
+; ------------------------
+;
+; Modifications history :
+; - June 2010: created by Lea Noreskal
+; - 28 June 2010: modified by Alain Coulais (N_PARAMS(), /help)
+;
+; - 2018-04-10 : AC 
+;  *0* input string should be a singleton !
+;  *1* manage repetition of same char ("  " --> "\ \ ")
+;  *2* remove of "." in the list ...
+;  *3* adding the space " " in the list (and remove bad 
+;                 STRJOIN(STRSPLIT(str_out, /EXTRACT), '\ ')  )
+;
+;- ------------------------
 ;
 function ESCAPE_SPECIAL_CHAR, str_in,  show_list=show_list, $
                               list_of_special_char=list_of_special_char, $
-                              test=test, help=help, verbose=verbose
+                              test=test, help=help, debug=debug, verbose=verbose
 ;
 if N_PARAMS() NE 1 then begin
-    MESSAGE, /continue, 'You must provide a input string !'
-    help=1
+   MESSAGE, /continue, 'You must provide a input string !'
+   help=1
 endif
 if KEYWORD_SET(help) then begin
-    print, 'function ESCAPE_SPECIAL_CHAR, str_in, show_list=show_list, $'
-    print, '           list_of_special_char=list_of_special_char, $'
-    print, '           test=test, help=help, verbose=verbose'
-    return, -1
+   print, 'function ESCAPE_SPECIAL_CHAR, str_in, show_list=show_list, $'
+   print, '           list_of_special_char=list_of_special_char, $'
+   print, '           test=test, help=help, debug=debug, verbose=verbose'
+   return, -1
 endif
 ;
+if N_ELEMENTS(str_in) GT 1 then MESSAGE, 'input string should not be an array'
+;
 if KEYWORD_SET(verbose) then begin
-    MESSAGE, /continue, '/verbose keyword not available, please contribute'
+   MESSAGE, /continue, '/verbose keyword not available, please contribute'
 endif
 ;
 ; list of special characteres can be modified if it is not complete
-;if( not KEYWORD_SET(list_of_special_char)) then list_of_special_char ='[*.^$,&\]'
+;if( not KEYWORD_SET(list_of_special_char)) then list_of_special_char ='[*^$,&\]'
```

Comment:
let's write a test for this change!

Reply:
those lines are commented ...
the real case in l47 vs l64 : removing "*" & ".", adding space " "

Reply:
as I said in the comments, the goal would be to *remove* this old code, and having in the C-code
using primitives (without "spawn")